Tests/scripts/common.py

```diff
@@ -260,67 +265,104 @@ def person_in_charge(commit):
+    current_pipeline_timestamp = parser.parse(current_pipeline.created_at)
+    return current_pipeline_timestamp > previous_pipeline_timestamp
+
 
+def is_pivot(current_pipeline: ProjectPipeline, previous_pipeline: ProjectPipeline)-> bool | None:
+    """
+    Is the current pipeline status a pivot from the previous pipeline status.
+    Args:
+        current_pipeline: The current pipeline object.
+        previous_pipeline: The previous pipeline object.   
     Returns:
-        boolean , the commit that triggered the current pipeline
+        True status changed from success to failed
+        False if the status changed from failed to success
+        None if the status didn't change or the pipelines are not in order of commits
     """
-    current_pipeline_commit_timestamp = None
-    previous_pipeline_commit_timestamp = None
-    the_triggering_commit = None
-    for commit in commits:
-        if commit.id == previous_pipeline_sha:
-            previous_pipeline_commit_timestamp = parser.parse(commit.created_at)
-        if commit.id == current_pipeline_sha:
-            current_pipeline_commit_timestamp = parser.parse(commit.created_at)
-            the_triggering_commit = commit
-    if not current_pipeline_commit_timestamp or not previous_pipeline_commit_timestamp:
-        return False, None
-    return current_pipeline_commit_timestamp > previous_pipeline_commit_timestamp, the_triggering_commit
-
-
-def is_pivot(single_pipeline_id, list_of_pipelines, commits):
+
+    in_order = are_pipelines_in_order(current_pipeline, previous_pipeline)
+    if in_order:
+        if previous_pipeline.status == 'success' and current_pipeline.status == 'failed':
+            return True
+        if previous_pipeline.status == 'failed' and current_pipeline.status == 'success':
+            return False
+    return None
+
+
+def get_reviewer(pr_url: str) -> str | None:
+    approved_reviewer = None
+    try:
+        # Extract the owner, repo, and pull request number from the URL
+        _1, _2,_3, repo_owner, repo_name,_4, pr_number = pr_url.split("/")  
+
+        # Get reviewers
+        reviews_url = f"https://api.github.com/repos/{repo_owner}/{repo_name}/pulls/{pr_number}/reviews"
+        reviews_response = requests.get(reviews_url, verify=False)
+        reviews_data = reviews_response.json()
+
+        # Find the reviewer who provided approval
+        for review in reviews_data:
+            if review["state"] == "APPROVED":
+                approved_reviewer = review["user"]["login"]
+                break
+    except Exception as e:
+        logging.error(f"Failed to get reviewer for PR {pr_url}: {e}")
+    return approved_reviewer
+
+
+def get_slack_user_name(name: str, name_mapping_path: str) -> str:
+    with open(name_mapping_path) as map:
+        mapping = json.load(map)
+    # If the name is 'github-actions[bot]' (docker image update bot reviewer)  return the owner of that bot.
+        if name == 'github-actions[bot]':
```

Comment:
move this `'github-actions[bot]'` to be a constant, as we did with the content-bot

Reply:
That make more sense. fixed.